### PR TITLE
lp leaderboard

### PR DESCRIPTION
**Pull request by szymcio32**

### CODE REVIEW

File: server/server/transform/events_transformer.py [hunk 1]
```diff
@@ -440,7 +434,9 @@ async def handle_swap(*args, **kwargs):
     await update_token_hour_data(db, token0, record['timestamp'], amount_total_USD_tracked, amount0_abs, fees_USD)
     await update_token_day_data(db, token1, record['timestamp'], amount_total_USD_tracked, amount1_abs, fees_USD)
     await update_token_hour_data(db, token1, record['timestamp'], amount_total_USD_tracked, amount1_abs, fees_USD)
-    
+
+    await calculate_volume_leaderboard_points(db, fees_USD, record)
```
Comment (princearoragithub) [inline server/server/transform/events_transformer.py:438]: This should be in the leaderboard transformer.

File: server/server/transform/leaderboard_transformer.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        }
+        await calculate_lp_leaderboard_points(event_data, db, rpc_url)
+        processed_positions_records += 1
+
+    logger.info(f'Successfully processed {processed_positions_records} Positions records for the leaderboard contest')
+
+
+async def calculate_lp_leaderboard_user_total_points(db: Database):
+    users_result = defaultdict(int)
+    async for position_record in yield_position_records(db, total_points_calculation=True):
```
Comment (princearoragithub) [inline server/server/transform/leaderboard_transformer.py:51]: This function can be done via mongo aggregation directly

Comment (princearoragithub) [reply server/server/transform/leaderboard_transformer.py:51]: Something like this: https://gist.github.com/princearoragithub/0108894cc3ff81923a9a59bdbf9ff8d8

File: server/server/transform/leaderboard_transformer.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        db_name = mongo_database.replace('-', '_')
+        db = mongo[db_name]
+        await handle_lp_leaderboard(db, rpc_url)
+        await calculate_lp_leaderboard_user_total_points(db)
+
+
+async def run_leaderboard_transformer(mongo_url: str, mongo_database: Database, rpc_url: str):
+    while True:
+        await process_leaderboard(mongo_url, mongo_database, rpc_url)
+        time.sleep(DAY_INTERVAL)
```
Comment (princearoragithub) [inline server/server/transform/leaderboard_transformer.py:88]: As you mentioned on discord, will figure out a better way to schedule this.

File: server/server/transform/lp_contest_updates.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+from structlog import get_logger
+
+
+logger = get_logger(__name__)
+
+
+TIME_VESTED_CONST = 1296000  # 15 days
+
+
+async def get_pool_boost(token0_address: str, token1_address) -> Decimal:
```
Comment (princearoragithub) [inline server/server/transform/lp_contest_updates.py:23]: Add type for token1_address

Comment (szymcio32) [reply server/server/transform/lp_contest_updates.py]: added

File: server/server/transform/positions_transformer.py [hunk 5]
```diff
@@ -1,82 +1,186 @@
...
+    del record['event']
+    logger.info("handle IncreaseLiquidity", **record)
+
+    token0 = await get_token_record(db, record['token0Address'], rpc_url)
+    token1 = await get_token_record(db, record['token1Address'], rpc_url)
+
+    amount0 = await amount_after_decimals(record['depositedToken0'], token0.get('decimals', DEFAULT_DECIMALS))
+    amount1 = await amount_after_decimals(record['depositedToken1'], token1.get('decimals', DEFAULT_DECIMALS))
+
+    await calculate_lp_leaderboard_points(record, db, rpc_url, Event.INCREASE_LIQUIDITY)
```
Comment (princearoragithub) [inline server/server/transform/positions_transformer.py:80]: Both here and in decrease liquidity, we should just store what we need to calculate lp leaderboard points. Actual calculation should happen only in leaderboard transformer (every 24 hours).

Comment (szymcio32) [reply server/server/transform/positions_transformer.py]: done

**Pull request merged**